src/engine.cairo

```diff
@@ -76,21 +76,21 @@ pub impl EngineImpl of EngineTrait {
         }
     }
 
-    fn pull_data(ref self: Engine, len: usize) -> ByteArray {
+    fn pull_data(ref self: Engine, len: usize) -> Result<ByteArray, felt252> {
         // TODO: check bounds with error handling
         let mut data = "";
         let mut i = self.opcode_idx + 1;
         let mut end = i + len;
         let script = *(self.scripts[self.script_idx]);
         if end > script.len() {
-            end = script.len();
+            return Result::Err(Error::SCRIPT_FAILED);
```

Comment:
I'd probably prefer this to be a new error type, maybe something like `SCRIPT_INVALID`. You can define it along with other errors in `src/errors.cairo`?